pymysql/connections.py

```diff
@@ -879,10 +879,11 @@ def _request_authentication(self):
             else:
                 authresp = b"\0"  # empty password
 
-        if self.server_capabilities & CLIENT.PLUGIN_AUTH_LENENC_CLIENT_DATA:
+        if (
+            self.server_capabilities & CLIENT.PLUGIN_AUTH_LENENC_CLIENT_DATA
+            or self.server_capabilities & CLIENT.SECURE_CONNECTION
+        ):
             data += _lenenc_int(len(authresp)) + authresp
-        elif self.server_capabilities & CLIENT.SECURE_CONNECTION:
-            data += struct.pack("B", len(authresp)) + authresp
```

Comment:
Don't do this. See https://dev.mysql.com/doc/internals/en/connection-phase-packets.html#packet-Protocol::HandshakeResponse

```
  if capabilities & CLIENT_PLUGIN_AUTH_LENENC_CLIENT_DATA {
lenenc-int     length of auth-response
string[n]      auth-response
  } else if capabilities & CLIENT_SECURE_CONNECTION {
1              length of auth-response
string[n]      auth-response
```

"length of auth-response" is 1byte, not lenenc when `capabilities & CLIENT_PLUGIN_AUTH_LENENC_CLIENT_DATA` is not true.

Reply:
Many excuses, Iʼve over-reacted without proper attention.
Iʼve pushed an updated variant which limits changes to the "client connection attributes" section.

Should I also add a test for this? The original issue can be triggered e.g. adding `program_name="x"*300` to connect() arguments (in this case, struct.pack() will fail), or cycle of connecting with program_name gradually increasing by 1 byte.

Reply:
If you can.  I love regression tests.  Otherwise, I may create a bug again.